core/state.go

```diff
@@ -406,15 +417,18 @@ func calculateContractCommitment(storageRoot, classHash, nonce *felt.Felt) *felt
 	return crypto.Pedersen(crypto.Pedersen(crypto.Pedersen(classHash, storageRoot), nonce), &felt.Zero)
 }
 
-func (s *State) updateDeclaredClasses(declaredClasses []DeclaredV1Class) error {
+func (s *State) updateDeclaredClasses(declaredClasses []DeclaredV1Class, revert bool) error {
 	classesTrie, classesCloser, err := s.classesTrie()
 	if err != nil {
 		return err
 	}
 
 	for _, declaredClass := range declaredClasses {
 		// https://docs.starknet.io/documentation/starknet_versions/upcoming_versions/#commitment
-		leafValue := crypto.Poseidon(leafVersion, declaredClass.CompiledClassHash)
+		leafValue := &felt.Zero
+		if !revert {
+			leafValue = crypto.Poseidon(leafVersion, declaredClass.CompiledClassHash)
```

Comment:
Does this set the leafValue to zero rather than whatever the old value was? Eg if we are reverting a declared class that had changed it's value from say '1' to '2', would this not revert it to '0' instead of '1'?

Reply:
Leafs in the classes trie are immutable, so they can only be set once to a non-zero value. So the old value is always zero.